presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -909,6 +909,14 @@ public void testFromInPredicateWithCastsAndNulls()
                 new InListExpression(ImmutableList.of(cast(toExpression(null, SMALLINT), BIGINT), toExpression(1L, SMALLINT))))));
     }
 
+    @Test
+    public void testFromInPredicateWithFunctions()
+    {
+        assertUnsupportedPredicate(equal(
+                function("date", C_TIMESTAMP.toSymbolReference()),
+                toExpression(DATE_VALUE, DATE)));
+    }
```

Comment:
I would like to see specific tests for that optimization with a loop short circuit. Also with and without complement.

Reply:
You know it because you implemented this case. I agree that it could be hard to guess from the test itself that it is target to short circuit. Surely, a comment there is needed for such test cases.

I am worried about the correctness for such cases (not the performance), today and in future. For performance a benchmark should be introduced.